### PR TITLE
Widgetize

### PR DESCRIPTION
**Pull request by baraabourghli**

### CODE REVIEW

File: javascript/est-widget.js [hunk 1]
```diff
@@ -28,24 +28,56 @@
...
+                var estURL = "https://api.govwizely.com/v2/environmental_solutions/search";
+
+                function estLoadData(search) {
+                    $.getJSON(composeURL(search), function (data) {
+                        $('#est-widget-result').html(syntaxHighlight(data));
+                    });
+                }
+
+                function composeURL(search) {
+                    return estURL + '?api_key=' + apiKey + '&q=' + search
```
Comment (calumhalcrow) [inline javascript/est-widget.js]: If `search` is an empty string, you cannot add `&q=` to the URL. The API doesn't behave as intended if you do. See the difference:

https://api.govwizely.com/environmental_solutions/search
https://api.govwizely.com/environmental_solutions/search?q=

I have [opened a bug](https://www.pivotaltracker.com/story/show/97957276) reporting this, but I think in general it's good to not pass query params when we're not actually doing a filtered search.

File: index.html [hunk 2]
```diff
@@ -3,9 +3,20 @@
   <head>
     <title>EST UI Widget</title>
     <link rel="stylesheet" type="text/css" href="style.css">
-    <script type="text/javascript" src="javascript/est-widget.js"></script>
+
+      <script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.3/jquery.min.js"></script>
```
Comment (calumhalcrow) [inline index.html:7]: Is it better to ask users to include JQuery, or should our code do it? (Doesn't the widget code [already do this](https://github.com/GovWizely/est-ui-widget/pull/2/files#diff-4220180f0c119d766886877485f1806aR4)?)

File: javascript/est-widget.js (hunk 1 above)
Comment (baraabourghli) [inline javascript/est-widget.js]: :+1:

File: index.html (hunk 2 above)
Comment (baraabourghli) [inline index.html:7]: I was thinking that [this code](https://github.com/GovWizely/est-ui-widget/pull/2/files#diff-4220180f0c119d766886877485f1806aR4) load the JQuery for use inside our anonymous function, and then by calling `window.jQuery.noConflict(true);` [we free up the JQuery](https://api.jquery.com/jquery.noconflict/) by passing the `true` to the noConflict call, we do this so we don't override the original version used in the website that embeds our widget.

But it seems like the JQuery tag is appended indeed in `<head>` but the browser is still complaining `Uncaught ReferenceError: $ is not defined`. i will look into this in more details.

Comment (baraabourghli) [inline index.html:7]: After trying many things, here is the conclusion:
[our code](https://github.com/GovWizely/est-ui-widget/pull/2/files#diff-4220180f0c119d766886877485f1806aR4) create the `script` tag and inject it correctly into `head`, but browsers do async source loading for the injected tag, and that is why in the following script, we can't use JQuery.

There are several ways to make the loading synchronous:
[1st example](https://stackoverflow.com/questions/2879509/dynamically-loading-javascript-synchronously)
[2nd example](https://stackoverflow.com/questions/8578617/inject-a-script-tag-with-remote-src-and-wait-for-it-to-execute/8578840#8578840)
but it is not recommended to do any synchronous loading.

I tried to solve it with [different ways](https://stackoverflow.com/questions/9899372/pure-javascript-equivalent-to-jquerys-ready-how-to-call-a-function-when-the), but all require the user to do some coding, or it require a fair amount of copy/paste.

What do you suggest in this case?

Comment (calumhalcrow) [inline index.html:7]: I see what the problem is now. ok, let's ask users to include jQuery before including our widget code. It's not a big problem TBH.

**Pull request merged**